src/System.IO.FileSystem/tests/FileInfo/get_LastWriteTime.cs

```diff
@@ -1,131 +1,45 @@
 
-using System;
-using System.Runtime.CompilerServices;
-using System.IO;
-using System.Collections;
-using System.Globalization;
-using System.Text;
-using System.Threading.Tasks;
 using Xunit;
 
-public class FileInfo_get_LastWriteTime
+namespace System.IO.FileSystem.Tests
 {
-    public static String s_strActiveBugNums = "";
-    public static String s_strDtTmVer = "2000/05/09 13:19";
-    public static String s_strClassMethod = "File.Directory()";
-    public static String s_strTFName = "get_LastWriteTime.cs";
-    public static String s_strTFPath = Directory.GetCurrentDirectory();
-
-    [Fact]
-    [OuterLoop]
-    public static void runTest()
+    public class FileInfo_get_LastWriteTime : FileSystemTest
     {
-        String strLoc = "Loc_000oo";
-        String strValue = String.Empty;
-        int iCountErrors = 0;
-        int iCountTestcases = 0;
-
-        try
+        [Fact]
+        public void WriteTimeRefreshes()
         {
-            /////////////////////////  START TESTS ////////////////////////////
-            ///////////////////////////////////////////////////////////////////
-
-
-            String filName = Path.Combine(TestInfo.CurrentDirectory, Path.GetRandomFileName());
-            FileInfo fil2;
-            Stream stream;
-
-
-            if (File.Exists(filName))
-                File.Delete(filName);
-
-
-
-            // [] Exception for null string
-            //-----------------------------------------------------------
-
-            // [] Create a directory and check the creation time
-
-            strLoc = "Loc_r8r7j";
-
-            new FileStream(filName, FileMode.Create).Dispose();
-            fil2 = new FileInfo(filName);
-            stream = fil2.OpenWrite();
-            stream.Write(new Byte[] { 10 }, 0, 1);
-            stream.Dispose();
-            Task.Delay(4000).Wait();
-            fil2.Refresh();
-            iCountTestcases++;
-            DateTime lastWriteTime = fil2.LastWriteTime;
-            if ((DateTime.Now - lastWriteTime).TotalMilliseconds < 2000 ||
-               (DateTime.Now - lastWriteTime).TotalMilliseconds > 5000)
+            string fileName = GetTestFilePath();
+            File.Create(fileName).Dispose();
+            FileInfo fileInfo = new FileInfo(fileName);
+            DateTime beforeWrite = DateTime.Now;
+            using (Stream stream = new FileInfo(fileName).OpenWrite())
             {
-                iCountErrors++;
-                Console.WriteLine(lastWriteTime);
-                Console.WriteLine((DateTime.Now - lastWriteTime).TotalMilliseconds);
-                printerr("Error_20hjx! Last Write Time time cannot be correct");
+                stream.Write(new Byte[] { 10 }, 0, 1);
             }
+            DateTime afterWrite = DateTime.Now;
+            fileInfo.Refresh();
+            DateTime lastWriteTime = fileInfo.LastWriteTime;
+            Assert.InRange<long>(fileInfo.LastWriteTime.Ticks, beforeWrite.Ticks, afterWrite.Ticks);
```

Comment:
This could be problematic on Linux (not sure about OSX).  Our current implementation uses utime, with a resolution of one second I believe.  If that's true, we could easily have a case where, for example, the last write time is at 1:23:45, the before time is 1:23:45.67 and the after time is 1:23:45.89... everything is correct, but the InRange will fail.

If I'm right that this is problematic, I'd suggest loosening the test a bit to allow some buffer.  Separately, you might open an issue and/or look to address the implementation to see about providing more accuracy, e.g. using utimes instead of utime, but we might want to wait for #2137 before doing that, as if memory serves it involves a structure that'll be very fragile to declare in managed code.